### PR TITLE
KLS-683 - Export academy signup (existing aventri user)

### PR DESCRIPTION
**Pull request by mforner13**
**Context**
This PR adds the first version of the export academy signup flow. In this version, the user is already registered for the export academy via Aventri, and has received a unique link where their email address will be pre-populated and they just need to add a password.

The backend implementation of the unique link has not yet been created, so the view currently relies on a registration uuid in the query parameters of the URL. Once the backend has been implemented this will be updated accordingly.

**Reusing IOO code**
This implementation reuses the logic used by international online offer's signup view which uses django, rather than the main signup flow using react. Currently

As there is a lot of common code between the IOO and EA signup flows, I have created a signup mixin and a verification code mixin in the `sso` app which both IOO and EA now use.

**To test**
- Make sure you have an EA registration for your chosen email in your local db
- Make sure you have directory-forms-api, directory-sso and directory-sso-proxy running locally
- Go to `export-academy/signup?registration-id=<your_registration_id>`
- Your email address should be pre-populated
- You should be able to go through the signup and verification code process and end up at `export-academy/events`
- If you already have an account in sso for your chosen email, try putting a `+1` before the @ sign

**Screenshots**
Screenshot 2023-06-21 at 10 37 24

Screenshot 2023-06-21 at 10 38 03

### Workflow
- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/KLS-683
- [x] Jira ticket has the correct status.
- [x] A clear/description pull request messaged added.

### Reviewing help

- [x] Explains how to test locally, including how to set up appropriate data
- [x] Includes screenshot(s) - ideally before and after, but at least after

### Housekeeping

- [x] Frontend assets have been re-compiled.

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: export_academy/templates/export_academy/accounts/base.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{% extends 'core/page_blank.html' %}
+{% block head_title %}UK Export Academy{% endblock %}
+{% block content %}
+<div id="sign-up" class="great">
+    <div class="great-signup">
+        <div class="great-signup__form-panel">
+            <a href="/" className="inline-block">
```
Comment (lewis-coulson-dit) [inline export_academy/templates/export_academy/accounts/base.html:7]: change `className` to `class`

File: export_academy/templates/export_academy/accounts/signup.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{% extends 'export_academy/accounts/base.html' %}
+{% block form %}
+    <h3 class="govuk-!-margin-top-3">Choose a password</h3>
+    <p class="govuk-!-margin-bottom-3">This is your UK Export Academy registered email address.</p>
+    {% csrf_token %}
+    <label class="form-label">Email address
```
Comment (lewis-coulson-dit) [inline export_academy/templates/export_academy/accounts/signup.html:6]: use a new govuk class here or a 'great' prefixed one

File: export_academy/templates/export_academy/accounts/signup.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    </div>
+    {% endfor %}
+    {% for field in form.hidden_fields %}
+        {{ field }}
+    {% endfor %}
+    <p class="govuk-!-margin-bottom-2">By signing up, you agree to our
+        <a class="govuk-link" href="/terms-and-conditions/">terms and conditions</a> and
+        <a class="govuk-link" href="/privacy-and-cookies/">privacy notice</a>
+    </p>
+    <button type="submit" class="button primary-button govuk-!-margin-top-6 great-full-width-button">Sign up</button>
```
Comment (lewis-coulson-dit) [inline export_academy/templates/export_academy/accounts/signup.html:21]: you should be able to just use `govuk-button` here

File: export_academy/templates/export_academy/accounts/signup.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    {% for field in form.visible_fields %}
+    <div class="govuk-form-group {% if field.errors %}govuk-form-group--error govuk-!-margin-top-6{% endif %}">
+        {% include 'domestic/contact/includes/govuk-form-field.html' with field=field %}
+    </div>
+    {% endfor %}
+    {% for field in form.hidden_fields %}
+        {{ field }}
+    {% endfor %}
+    <p class="govuk-!-margin-bottom-2">By signing up, you agree to our
+        <a class="govuk-link" href="/terms-and-conditions/">terms and conditions</a> and
```
Comment (stuart-madetech) [inline export_academy/templates/export_academy/accounts/signup.html:18]: Do these need to be variables?

File: export_academy/templates/export_academy/accounts/verification_code.html [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    </p>
+    <p class="govuk-!-margin-bottom-0">Enter the code below to complete registration.</p>
+    {% csrf_token %}
+    {% comment %} {{ form }} {% endcomment %}
+    {% for field in form.visible_fields %}
+    <div class="govuk-form-group govuk-!-margin-top-6 {% if field.errors %}govuk-form-group--error{% endif %}">
+        {% include 'domestic/contact/includes/govuk-form-field.html' with field=field %}
+    </div>
+    {% endfor %}
+    <button type="submit" class="button primary-button govuk-!-margin-top-6 great-full-width-button">Submit</button>
```
Comment (lewis-coulson-dit) [inline export_academy/templates/export_academy/accounts/verification_code.html:15]: same as above comment

File: styles/great/_sign-up.scss [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        .button {
+            min-width: 230px;
+        }
+
+        &__confirmation {
+            .fa-info-circle {
+                font-size: 40px;
+            }
+
+            .form-label {
```
Comment (lewis-coulson-dit) [inline styles/great/_sign-up.scss:146]: change in line with above comment

File: styles/great/_sign-up.scss [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+                }
+
+                i {
+                    margin-right: 30px;
+                    font-size: 1.5em;
+                }
+            }
+        }
+
+        .button {
```
Comment (lewis-coulson-dit) [inline styles/great/_sign-up.scss:137]: change in line with above comment

**Review by stuart-madetech — APPROVED** [review]
LGTM - only one comment around urls being set but assume its like that for a reason

**Pull request merged**